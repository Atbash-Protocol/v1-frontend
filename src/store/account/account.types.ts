import { StaticJsonRpcProvider, JsonRpcProvider } from '@ethersproject/providers';
import { Networks } from 'constants/index';
import { Bond } from 'helpers/bond/bond';
import { IToken } from 'helpers/tokens';

export interface IGetBalances {
    address: string;
    networkID: Networks;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
}

export interface IAccountBalances {
    balances: {
        wsBASH: string;
        sBASH: string;
        BASH: string;
        aBash: string;
    };
}

export interface ILoadAccountDetails {
    address: string;
    networkID: Networks;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
}

export interface IUserAccountDetails {
    balances: {
        BASH: string;
        sBASH: string;
        wsBASH: string;
        aBash: string;
    };
    redeeming: {
        BASH: number;
        aBash: number;
    };
    staking: {
        BASH: number;
        sBASH: number;
    };
    wrapping: {
        sBASHAllowance: number;
    };
}

export interface ICalcUserBondDetails {
    address: string;
    bond: Bond;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
    networkID: Networks;
}

export interface ICalcUserTokenDetails {
    address: string;
    token: IToken;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
    networkID: Networks;
}

export interface IAccountSlice {
    bonds: { [key: string]: IUserBondDetails };
    balances: {
        wsBASH: string;
        sBASH: string;
        BASH: string;
    };
    loading: boolean;
    redeeming: {
        BASH: number;
    };
    staking: {
        BASH: number;
        sBASH: number;
    };
    wrapping: {
        sBASHAllowance: number;
    };
    tokens: { [key: string]: IUserTokenDetails };
}

export interface IUserTokenDetails {
    allowance: number;
    balance: number;
    isAvax?: boolean;
}

export interface IUserBondDetails {
    allowance: number;
    balance: number;
    avaxBalance: number;
    interestDue: number;
    bondMaturationBlock: number;
    pendingPayout: number; //Payout formatted in gwei.
}
