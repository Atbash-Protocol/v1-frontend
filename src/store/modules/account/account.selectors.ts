import Decimal from 'decimal.js';
import { utils } from 'ethers';
import { createSelector } from 'reselect';

import { RootState } from 'store/store';

export const selectBASHBalance = (state: RootState): Decimal => {
    const BASHAmount = state.account.balances.BASH; // 9 Decimals

    return new Decimal(BASHAmount.toString()).div(10 ** 9);
};

export const selectSBASHBalance = (state: RootState): Decimal => {
    const SBASHAmount = state.account.balances.SBASH; // 9 Decimals

    return new Decimal(SBASHAmount.toString()).div(10 ** 9);
};

export const selectWSBASHBalance = (state: RootState): Decimal => {
    const WSBASHAmount = state.account.balances.WSBASH;

    return new Decimal(WSBASHAmount.toString()).div(10 ** 9); // 9 Decimals
};

export const selectFormattedStakeBalance = createSelector([selectBASHBalance, selectSBASHBalance, selectWSBASHBalance], (BASHBalance, SBASHBalance, WSBASHBalance) => {
    return Object.entries({ BASH: BASHBalance, SBASH: SBASHBalance, WSBASH: WSBASHBalance }).reduce(
        (acc, [key, amount]) => {
            return {
                ...acc,
                [key]: [amount.toFixed(2), key.toLocaleUpperCase()].join(' '),
            };
        },
        {
            BASH: '0.00 BASH',
            SBASH: '0.00 SBASH',
            WSBASH: '0.00 WSBASH',
        },
    );
});

export const selectFormattedBASHBalance = (state: RootState): string => {
    const BASHAmount = state.account.balances.BASH; // 9 Decimals

    return [Number(utils.formatUnits(BASHAmount, 'gwei')).toFixed(2), 'BASH'].join(' ');
};

export const selectUserStakingAllowance = (state: RootState) => {
    const { BASH, SBASH } = state.account.stakingAllowance;

    return {
        BASHAllowanceNeeded: BASH.eq(0),
        SBASHAllowanceNeeded: SBASH.eq(0),
    };
};

export const selectAccountLoading = (state: RootState): boolean => state.account.loading;
