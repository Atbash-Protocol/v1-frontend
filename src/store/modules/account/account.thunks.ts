import { JsonRpcProvider } from "@ethersproject/providers";
import { createAsyncThunk } from "@reduxjs/toolkit";
import { MemoTokenContract, TimeTokenContract } from "abi";
import { getAddresses } from "constants/addresses";
import { ethers } from "ethers";
import { IReduxState } from "store/slices/state.interface";

export const loadBalancesAndAllowances = createAsyncThunk("account/loadBalancesAndAllowances", async ({ chainID, provider, address }: any, { getState }): Promise<any> => {
    let BASHbalance = 0;
    let sBASHBalance = 0;
    let wsBASHBalance = 0;
    let stakeAllowance = 0;
    let unstakeAllowance = 0;
    let wrapAllowance = 0;
    let redeemAllowance = 0;

    const {
        main: {
            contracts: { BASH_CONTRACT, SBASH_CONTRACT, STAKING_ADDRESS, STAKING_HELPER_ADDRESS, WSBASH_ADDRESS },
        },
    } = getState() as IReduxState;

    if (BASH_CONTRACT) {
        BASHbalance = await BASH_CONTRACT.balanceOf(address);
        stakeAllowance = await BASH_CONTRACT.allowance(address, STAKING_HELPER_ADDRESS?.address);
        // disable: redeemAllowance = await sbContract.allowance(address, addresses.REDEEM_ADDRESS);
    }

    if (SBASH_CONTRACT) {
        try {
            wrapAllowance = await SBASH_CONTRACT.allowance(address, WSBASH_ADDRESS?.address);
            unstakeAllowance = await SBASH_CONTRACT.allowance(address, STAKING_ADDRESS?.address);
        } catch (err) {
            console.log(err);
        }
    }

    // if (addresses.WSBASH_ADDRESS) {
    //     const wsBASHContract = new ethers.Contract(addresses.WSBASH_ADDRESS, SBashTokenContract, provider);
    //     wsBASHBalance = await wsBASHContract.balanceOf(address);
    // }

    return {
        balances: {
            WSBASH: ethers.utils.formatEther(wsBASHBalance),
            SBASH: ethers.utils.formatUnits(sBASHBalance, "gwei"),
            BASH: ethers.utils.formatUnits(BASHbalance, "gwei"),
        },

        stakingAllowance: {
            BASH: Number(stakeAllowance),
            SBASH: Number(unstakeAllowance),
        },
        // wrapping: {
        //     sBASHAllowance: Number(wrapAllowance),
        // },
    };
});
