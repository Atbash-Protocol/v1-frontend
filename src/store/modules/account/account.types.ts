export interface AccountSlice {
    stakingAllowance: {
        BASH: number;
        SBASH: number;
    };
    balances: {
        BASH: number;
        SBASH: number;
        WSBASH: number;
    };
    loading: boolean;
}
