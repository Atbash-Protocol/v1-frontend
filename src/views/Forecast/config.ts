export const MIN_DAYS = 1;
export const MAX_DAYS = 365;
export const LAMBO_PRICE = 222_004;
export const EPOCH_PER_DAY = 3;
