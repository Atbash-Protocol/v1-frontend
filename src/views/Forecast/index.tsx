import { Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';

import { BCard } from 'components/BCard';
import { theme } from 'constants/theme';

import ForecastMetrics from './components/ForecastMetrics';
import Form from './components/Form';

function Forecast() {
    const { t } = useTranslation();

    return (
        <BCard title={t('globe:ForecastTitle')} zoom={true}>
            <Typography sx={{ color: theme.palette.primary.light }}>
                <>{t('globe:EstimateYourReturns')} </>
            </Typography>
            <Typography sx={{ color: theme.palette.primary.light, mt: theme.spacing(2) }}>
                <>{t('globe:ForecastWarning')} </>
            </Typography>

            <ForecastMetrics />

            <Form />
        </BCard>
    );
}

export default Forecast;
