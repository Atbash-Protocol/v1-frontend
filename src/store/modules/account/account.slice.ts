import { createSlice } from "@reduxjs/toolkit";
import { BigNumber } from "ethers";
import { loadBalancesAndAllowances } from "./account.thunks";
import { AccountSlice } from "./account.types";

const initialState: AccountSlice = {
    balances: {
        BASH: BigNumber.from(0),
        SBASH: BigNumber.from(0),
        WSBASH: BigNumber.from(0),
    },
    stakingAllowance: {
        BASH: BigNumber.from(0),
        SBASH: BigNumber.from(0),
    },
    loading: true,
};

export const marketSlice = createSlice({
    name: "app-account-slice",
    initialState,
    reducers: {},
    extraReducers: builder => {
        builder.addCase(loadBalancesAndAllowances.fulfilled, (state, action) => {
            return {
                ...state,
                ...action.payload,
                loading: false,
            };
        });
        builder.addCase(loadBalancesAndAllowances.rejected, (state, action) => {
            return {
                ...state,
                loading: false,
            };
        });
    },
});

export default marketSlice.reducer;
