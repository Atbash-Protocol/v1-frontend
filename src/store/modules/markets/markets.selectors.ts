import Decimal from 'decimal.js';
import { utils } from 'ethers';
import { createSelector } from 'reselect';

import { formatUSDFromDecimal } from 'helpers/price-units';
import { RootState } from 'store/store';

import { selectBASHBalance } from '../account/account.selectors';
import { selectReserve } from '../app/app.selectors';

export const selectFormattedBashBalance = (state: RootState): string | null => {
    const { dai } = state.markets.markets;
    const balance = selectBASHBalance(state);

    if (!dai) return null;

    return formatUSDFromDecimal(balance.mul(dai), 2);
};

export const selectMarketsLoading = (state: RootState): boolean => {
    const { loading } = state.markets;

    return loading;
};

export const selectDaiPrice = (state: RootState): Decimal => {
    return new Decimal(state.markets.markets.dai ?? 0);
};

export const selectMarketPrice = createSelector([selectReserve, selectDaiPrice], (reserve, dai) => reserve.mul(dai));

// converted marketPrice from wei
export const selectComputedMarketPrice = createSelector([selectMarketPrice], marketPrice => marketPrice.div(10 ** 9));
