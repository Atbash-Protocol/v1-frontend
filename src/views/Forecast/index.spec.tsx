import { render, screen } from '@testing-library/react';
import Decimal from 'decimal.js';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';

import * as AccountSelectorModule from 'store/modules/account/account.selectors';
import * as MarketsSelectorModule from 'store/modules/markets/markets.selectors';
import * as MetricsSelectorModule from 'store/modules/metrics/metrics.selectors';

import Forecast from '.';

jest.mock('./components/Dashboard/components/ForecastMetrics', () => () => <>Forecast Metrics</>);
jest.mock('./components/Dashboard', () => (props: any) => <>Dashboard {JSON.stringify(props)}</>);

describe('Forecast', () => {
    it('renders', async () => {
        jest.spyOn(AccountSelectorModule, 'selectSBASHBalance').mockReturnValue(new Decimal(10.93));
        jest.spyOn(MarketsSelectorModule, 'selectComputedMarketPrice').mockReturnValue(new Decimal(82.23));
        jest.spyOn(MetricsSelectorModule, 'selectStakingRebasePercentage').mockReturnValue(new Decimal(20));

        const mockStore = createMockStore([]);
        const store = mockStore({});

        const comp = render(
            <Provider store={store}>
                <Forecast />
            </Provider>,
        );

        expect(comp).toMatchSnapshot();
    });
});
