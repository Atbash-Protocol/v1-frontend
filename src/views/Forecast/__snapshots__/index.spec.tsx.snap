// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-8fs7kz"
      >
        <div
          class="MuiBox-root css-19na2ny"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
          >
            globe:ForecastTitle
          </h4>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-15j5nkq-MuiTypography-root"
        >
          globe:EstimateYourReturns
           
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-18y7bwq-MuiTypography-root"
        >
          globe:ForecastWarning
           
        </p>
        Forecast Metrics
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                sBASH AMOUNT
              </p>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  inputmode="numeric"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  type="number"
                  value="0"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <div
                    class="MuiBox-root css-og7htl"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      Max
                    </p>
                  </div>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                REWARD YIELD (%)
              </p>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  inputmode="numeric"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  type="number"
                  value="0"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <div
                    class="MuiBox-root css-og7htl"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      Max
                    </p>
                  </div>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
              >
                BASH Price AT PURCHASE ($)
              </p>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  inputmode="numeric"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  type="number"
                  value="0"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <div
                    class="MuiBox-root css-og7htl"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      Max
                    </p>
                  </div>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Future BASH MARKET PRICE ($)
              </p>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  inputmode="numeric"
                  pattern="^[0-9]*[.,]?[0-9]*$"
                  type="number"
                  value="0"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
                >
                  <div
                    class="MuiBox-root css-og7htl"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    >
                      Max
                    </p>
                  </div>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1unhm3"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1pzpssa-MuiTypography-root"
              >
                common:day_other
              </p>
              <span
                class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root MuiSlider-vertical css-y50uh2-MuiSlider-root"
              >
                <span
                  class="MuiSlider-rail css-qx5f9e-MuiSlider-rail"
                />
                <span
                  class="MuiSlider-track css-ada0pd-MuiSlider-track"
                  style="bottom: 0%; height: 0%;"
                />
                <span
                  class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-y35fuw-MuiSlider-thumb"
                  data-index="0"
                  style="bottom: 0%;"
                >
                  <input
                    aria-orientation="vertical"
                    aria-valuemax="365"
                    aria-valuemin="0"
                    aria-valuenow="0"
                    data-index="0"
                    max="365"
                    min="0"
                    step="1"
                    style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                    type="range"
                    value="0"
                  />
                  <span
                    aria-hidden="true"
                    class="MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
                  >
                    <span
                      class="MuiSlider-valueLabelCircle"
                    >
                      <span
                        class="MuiSlider-valueLabelLabel"
                      >
                        0
                      </span>
                    </span>
                  </span>
                </span>
              </span>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          >
            <span>
              Initial investment
            </span>
            <span>
              Current wealth
            </span>
            <span>
              Bash rewards estimation
            </span>
            <span>
              Potential return
            </span>
            <span>
              Lambo prices
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-8fs7kz"
    >
      <div
        class="MuiBox-root css-19na2ny"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1b3l217-MuiTypography-root"
        >
          globe:ForecastTitle
        </h4>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-15j5nkq-MuiTypography-root"
      >
        globe:EstimateYourReturns
         
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-18y7bwq-MuiTypography-root"
      >
        globe:ForecastWarning
         
      </p>
      Forecast Metrics
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              sBASH AMOUNT
            </p>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                inputmode="numeric"
                pattern="^[0-9]*[.,]?[0-9]*$"
                type="number"
                value="0"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <div
                  class="MuiBox-root css-og7htl"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Max
                  </p>
                </div>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              REWARD YIELD (%)
            </p>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                inputmode="numeric"
                pattern="^[0-9]*[.,]?[0-9]*$"
                type="number"
                value="0"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <div
                  class="MuiBox-root css-og7htl"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Max
                  </p>
                </div>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
            >
              BASH Price AT PURCHASE ($)
            </p>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                inputmode="numeric"
                pattern="^[0-9]*[.,]?[0-9]*$"
                type="number"
                value="0"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <div
                  class="MuiBox-root css-og7htl"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Max
                  </p>
                </div>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Future BASH MARKET PRICE ($)
            </p>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-1facavo-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                inputmode="numeric"
                pattern="^[0-9]*[.,]?[0-9]*$"
                type="number"
                value="0"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
              >
                <div
                  class="MuiBox-root css-og7htl"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    Max
                  </p>
                </div>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1unhm3"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1pzpssa-MuiTypography-root"
            >
              common:day_other
            </p>
            <span
              class="MuiSlider-colorPrimary MuiSlider-sizeMedium MuiSlider-root MuiSlider-vertical css-y50uh2-MuiSlider-root"
            >
              <span
                class="MuiSlider-rail css-qx5f9e-MuiSlider-rail"
              />
              <span
                class="MuiSlider-track css-ada0pd-MuiSlider-track"
                style="bottom: 0%; height: 0%;"
              />
              <span
                class="MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium MuiSlider-thumb css-y35fuw-MuiSlider-thumb"
                data-index="0"
                style="bottom: 0%;"
              >
                <input
                  aria-orientation="vertical"
                  aria-valuemax="365"
                  aria-valuemin="0"
                  aria-valuenow="0"
                  data-index="0"
                  max="365"
                  min="0"
                  step="1"
                  style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                  type="range"
                  value="0"
                />
                <span
                  aria-hidden="true"
                  class="MuiSlider-valueLabel css-1kz0hui-MuiSlider-valueLabel"
                >
                  <span
                    class="MuiSlider-valueLabelCircle"
                  >
                    <span
                      class="MuiSlider-valueLabelLabel"
                    >
                      0
                    </span>
                  </span>
                </span>
              </span>
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
        >
          <span>
            Initial investment
          </span>
          <span>
            Current wealth
          </span>
          <span>
            Bash rewards estimation
          </span>
          <span>
            Potential return
          </span>
          <span>
            Lambo prices
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
